modcc/module.cpp

```diff
@@ -791,10 +791,10 @@ int Module::semantic_func_proc() {
                 //      ll1_ = 2+x
                 //      ll0_ = ll1_*(y + 1)
                 //      a = 2 + ll0_
-                for(auto e=b.begin(); e!=b.end(); ++e) {
-                    if(auto ass = (*e)->is_assignment()) {
+                for(auto& e: b) {
+                    if(auto ass = e->is_assignment()) {
                         if(ass->rhs()->is_function_call()) {
-                            ass->replace_rhs(inline_function_call(ass->rhs()));
+                            e = inline_function_call(e);
```

Comment:
`inline_function_call` will return either
* an assigment expression; or
* an if-else expression

Right?

Reply:
Right. It will take whatever expression is in the function (if more than one expression is in the body of the function it will return an error), it checks if the expression is an if expression or an assignment expression: if yes, it will modify the expression and return the inlined version; if no it will return an error.